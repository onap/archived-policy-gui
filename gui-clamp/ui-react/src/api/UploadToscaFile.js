/*
 * -
 *  * ============LICENSE_START=======================================================
 *  *  Copyright (C) 2021 Nordix Foundation.
 *  * ================================================================================
 *  * Licensed under the Apache License, Version 2.0 (the "License");
 *  * you may not use this file except in compliance with the License.
 *  * You may obtain a copy of the License at
 *  *
 *  *      http://www.apache.org/licenses/LICENSE-2.0
 *  *
 *  * Unless required by applicable law or agreed to in writing, software
 *  * distributed under the License is distributed on an "AS IS" BASIS,
 *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  * See the License for the specific language governing permissions and
 *  * limitations under the License.
 *  *
 *  * SPDX-License-Identifier: Apache-2.0
 *  * ============LICENSE_END=========================================================
 *
 */
import Button from "react-bootstrap/Button";
import React, { useState } from "react";

const UploadToscaFile = (props) => {
  const [windowLocationPathName, setWindowLocationPathname] = useState('');

  const postServiceTemplateHandler = async (event) => {

    event.preventDefault();
    setWindowLocationPathname(window.location.pathname);

    const response = await fetch(windowLocationPathName +
      '/restservices/clds/v2/toscaControlLoop/commissionToscaTemplate', {
      method: 'POST',
      headers: {
        "Content-Type": "application/json"
      },
      credentials: 'same-origin',
      body: JSON.stringify(props.toscaObject),
    });

    const responseObj = await response;
    const responseMessage = await response.text();

    props.onResponseReceived(responseObj, responseMessage);

  }

  return (
    <React.Fragment>
      <Button variant="primary"
              block={ true }
              type="submit"
              onClick={ postServiceTemplateHandler }>
        Upload Tosca Service Template
      </Button>
    </React.Fragment>
  );

};

export default UploadToscaFile;
