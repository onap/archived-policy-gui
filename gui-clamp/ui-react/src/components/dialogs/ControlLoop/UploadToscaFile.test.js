/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import UploadToscaFile from './UploadToscaFile';
import toJson from "enzyme-to-json";
import { act } from "react-dom/test-utils";

describe('Verify UploadToscaFile', () => {

  it("renders without crashing", () => {
    shallow(<UploadToscaFile/>);
  });

  it("renders correctly", () => {
    const tree = shallow(<UploadToscaFile/>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should have a Button element', () => {
    const container = shallow(<UploadToscaFile/>)
    expect(container.find('Button').length).toEqual(1);
  });

  it('button should call postServiceTemplateHandler when clicked', async () => {
    const component = mount(<UploadToscaFile/>)
    const logSpy = jest.spyOn(console, 'log');
    const event = {
      preventDefault() {
      }
    };

    act(async () => {
      component.find('[variant="primary"]').get(0).props.onClick(event);
      expect(logSpy).toHaveBeenCalledWith('postServiceTemplateHandler called');
    })

  });

  it('should have a Button element with specified text', () => {
    const container = shallow(<UploadToscaFile/>)
    expect(container.find('Button').text()).toBe('Upload Tosca Service Template');
  });
});
