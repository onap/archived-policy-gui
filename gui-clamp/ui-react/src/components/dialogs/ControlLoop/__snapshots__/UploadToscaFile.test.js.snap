// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify UploadToscaFile renders correctly 1`] = `
<Fragment>
  <Button
    active={false}
    block={true}
    disabled={false}
    onClick={[Function]}
    type="submit"
    variant="primary"
  >
    Upload Tosca Service Template
  </Button>
</Fragment>
`;
